cmd/nerdctl/container_run_soci_linux_test.go

```diff
@@ -0,0 +1,110 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/containerd/nerdctl/pkg/testutil"
+)
+
+func TestRunSoci(t *testing.T) {
+	testutil.DockerIncompatible(t)
+	base := testutil.NewBase(t)
+	requiresSoci(base)
+
+	//counting intial snapshot mounts
+	base.T.Logf("counting expected mounts") //DEBUG
+	initialMounts, err := exec.Command("mount").Output()
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	remoteSnapshotsInitialCount := strings.Count(string(initialMounts), "fuse.rawBridge")
+	base.T.Logf("number of initial mounts: %v", remoteSnapshotsInitialCount)
+
+	if remoteSnapshotsInitialCount != 0 {
+		t.Fatalf("initial mounts count isn't zero")
+	}
+
+	//calling nerdctl run with '--snapshotter=soci ...'
+	runOutput := base.Cmd("--snapshotter=soci", "run", "--rm", testutil.FfmpegSociImage).Out()
+	base.T.Logf("run output: %s", runOutput) //DEBUG
+
+	//count actual mounts
+	actualMounts, err := exec.Command("mount").Output()
+	remoteSnapshotsActualCount := strings.Count(string(actualMounts), "fuse.rawBridge")
+	base.T.Logf("number of actual mounts: %v", remoteSnapshotsActualCount)
+
+	//clean up image
+	stopOutput = base.Cmd("stop", testutil.FfmpegSociContainerTag).Out()
```

Comment:
Does this work? You might want to specify --name in run and use container name to stop it? You may also need `rm` to clean up the container.

Reply:
Since we run the container with the `--rm` flag shouldn't it get cleaned up after we stop/exit it?

Reply:
You are right we don't need additional `rm` here.